src/WaveTrack.cpp

```diff
@@ -1260,6 +1261,9 @@ void WaveTrack::Paste(double t0, const Track *src)
 {
    bool editClipCanMove = GetEditClipsCanMove();
 
+   auto pastingFromTempTrack = !src->GetOwner();
+   auto pastingFromClipboard = src->GetOwner() && src->GetOwner().get() == &Clipboard::Get().GetTracks();
```

Comment:
There is code here for detection of the origin of the clip.  But can there be another fix that sets the clip names right when they are made?

Could you instead correct the names in the OnCut and OnCopy functions in EditMenus.cpp?

Reply:
Empty string is valid clip name, so I guess we cannot guess why it's empty

Reply:
Perhaps then, a clip could store a std::optional<wxString> !  Or otherwise invent a setter method for validity of the name.